### PR TITLE
Add better logs error messages

### PR DESCRIPTION
**Pull request by dchw**

### CODE REVIEW

File: cmd/earthly/main.go [hunk 1]
```diff
@@ -2830,12 +2830,14 @@ func (app *earthlyApp) actionBuildImp(c *cli.Context, flagArgs, nonFlagArgs []st
...
 				if err != nil {
+					err := errors.Wrapf(err, "failed to write log to disk")
 					app.console.Warnf(err.Error())
 					return
 				}
 
 				id, err := cc.UploadLog(logPath)
 				if err != nil {
+					err := errors.Wrapf(err, "failed to upload log")
 					app.console.Warnf(err.Error())
```
Comment (brandonSc) [inline cmd/earthly/main.go:2841]: Wonder if this should check for unauthorized explicitly and log an info level instead. (i.e. getting a red error message even though the build succeeded looks misleading)

**Comment by brandonSc** [comment]
ET-74

^ will this trigger Linear.app integration to clue in?

**Comment by dchw** [comment]
No. Linear is not hooked up to this repo currently.

File: cmd/earthly/main.go [hunk 2]
```diff
@@ -2821,7 +2821,7 @@ func (app *earthlyApp) actionBuildImp(c *cli.Context, flagArgs, nonFlagArgs []st
 	// Default upload logs, unless explicitly configured
 	if !app.cfg.Global.DisableLogSharing {
 		_, _, _, whoAmIErr := cc.WhoAmI()
-		isLoggedIn := err == nil
+		isLoggedIn := whoAmIErr == nil
```
Comment (dchw) [inline cmd/earthly/main.go:2824]: @brandonSc This is why you managed to get an unauthorized. Fixed now.

Comment (brandonSc) [reply cmd/earthly/main.go:2824]: So we shouldn't even get the error when not logged in. I think that invalidates my other comment below.

Comment (dchw) [reply cmd/earthly/main.go:2824]: Yeah. I was super confused and then realized my mistake :facepalm:

**Pull request merged**